File: electron/database/swwwConfig.ts
import {
  ResizeType,
  FilterType,
  TransitionType,
  transitionPosition
} from '../../src/hooks/swwwConfigStore'

export const initialSwwwConfigDB = {
  resizeType: ResizeType.crop,
  fillColor: '#000000',
  filterType: FilterType.Lanczos3,
  transitionType: TransitionType.simple,
  transitionStep: 90,
  transitionDuration: 3,
  transitionFPS: 60,
  transitionAngle: 45,
  transitionPositionType: 'alias',
  transitionPosition: transitionPosition.center,
  transitionPositionIntX: 960,
  transitionPositionIntY: 540,
  transitionPositionFloatX: 0.5,
  transitionPositionFloatY: 0.5,
  invertY: 0 , // Same as false
  transitionBezier: '.25,.1,.25,1',
  transitionWaveX: 20,
  transitionWaveY: 20
}

export type swwwConfig = typeof initialSwwwConfigDB

File: daemon-go/internal/monitor/swww_config.go
package monitor

// SwwwConfig holds the configuration for the swww daemon.
type SwwwConfig struct {
	ResizeType             string  `json:"resizeType"`
	FillColor              string  `json:"fillColor"`
	FilterType             string  `json:"filterType"`
	TransitionType         string  `json:"transitionType"`
	TransitionStep         int     `json:"transitionStep"`
	TransitionDuration     float64 `json:"transitionDuration"`
	TransitionFPS          int     `json:"transitionFPS"`
	TransitionAngle        float64 `json:"transitionAngle"`
	TransitionPositionType string  `json:"transitionPositionType"`
	TransitionPosition     string  `json:"transitionPosition"`
	TransitionPositionIntX int     `json:"transitionPositionIntX"`
	TransitionPositionIntY int     `json:"transitionPositionIntY"`
	TransitionPositionFloatX float64 `json:"transitionPositionFloatX"`
	TransitionPositionFloatY float64 `json:"transitionPositionFloatY"`
	InvertY                bool    `json:"invertY"`
	TransitionBezier       string  `json:"transitionBezier"`
	TransitionWaveX        int     `json:"transitionWaveX"`
	TransitionWaveY        int     `json:"transitionWaveY"`
}
